src/stim/mem/simd_bit_table.inl

```diff
@@ -90,12 +90,28 @@ simd_bit_table<W> simd_bit_table<W>::inverse_assuming_lower_triangular(size_t n)
     return result;
 }
 
+/// SIMD transpose:
+/// Suppose we have a matrix with minor axis contiguous in memory
+///
+/// A00  A01  ...  A0n
+/// A10  A11  ...  A1n
+///           ...
+/// An0  An1  ...  A2n
+///
+/// Aij consists of W words (W * W bits), and it is stored in memory
```

Comment:
Ah, okay, those `A` values aren't actually contiguous within themselves and actually refer to square blocks of bits. That is *very* unclear from the diagram above, though. The diagram above appears to be emphasizing that each Axy is contiguous and then contiguous with Ax(y+1).

The way I think about these operations is that each bit in the full address (down to the bit, not just down to the word) is associated with bits from the major index and bits from the minor index (again down to the bit, not down to the word). What transposing does is to adjust this mapping, so that e.g. instead of minor axis bit 0 mapping to address bit 0 the major axis bit 0 will map to address bit 0. But this may only make sense to me because I'm familiar with quantum circuit diagrams, where qubit swap operations correspond to exactly this kind of remapping within the state vector.

The key thing to realize, from an "I'm implementing bit swaps within the address bits" perspective, is that any permutation of the bits whose positions are larger than log_2(W) can be achieved via `std::swap`. But for bit positions below log_2(W) you have to start doing work within the words, and this involves things like masking and shifting within a word or between two words.